src/app/plotly/plotly.component.ts

```diff
@@ -58,21 +58,23 @@ export class PlotlyComponent implements OnInit, AfterViewInit, OnChanges {
       console.log(this.data);
       console.log('[PlotlyComponent] layout: ');
       console.log(this.layout);
-      const div = document.getElementById('plotly-' + this.randomId);
-      Plotly.deleteTraces(div, 0);
-      Plotly.addTraces(div, this.data);
+      this.createPlot();
```

Comment:
Don't we need to delete the plot data somehow?

Reply:
No, newPlot remove and create new plot, but in newer version of plotly is function react which is more efficient than newPlot (https://plot.ly/javascript/plotlyjs-function-reference/#plotlyreact) so I added it.

Reply:
According to the documentation:
Plotly.react has the same signature as Plotly.newPlot above, and can be used in its place to **create a plot**, but when called again on the same <div> will update it far more efficiently than Plotly.newPlot